libres/tests/analysis/ies/test_ies_enkf_main.cpp

```diff
@@ -46,43 +43,135 @@ TEST_CASE("ies_enkf_linalg_extract_active", "[analysis]") {
+                         nullptr, nullptr, nullptr);
+    ies_enkf_data_update_ens_mask(data, ens_mask);
+
+    matrix_type *W0 = matrix_alloc(ens_size, ens_size);
+    for (int i = 0; i < ens_size; i++) {
+        for (int j = 0; j < ens_size; j++)
+            matrix_iset(W0, i, j, i * ens_size + j);
+    }
+    ies_enkf_linalg_store_active_W(data, W0);
+
+    {
+        matrix_type *W = ies_enkf_alloc_activeW(data);
+        REQUIRE(matrix_equal(W, W0));
+        matrix_free(W);
+    }
 
-    matrix_type *E_obs_deactivate = matrix_alloc(state_size, ens_size);
-    ies_enkf_linalg_extract_active(data, E_obs_deactivate, stdout, false);
+    // Deactivate one realization
+    bool_vector_iset(ens_mask, 1, false);
+    ies_enkf_data_update_ens_mask(data, ens_mask);
+    {
+        matrix_type *W = ies_enkf_alloc_activeW(data);
+        for (int i = 0; i < ens_size - 1; i++) {
+            for (int j = 0; j < ens_size - 1; j++) {
+                int i0 = i + (i > 0);
+                int j0 = j + (j > 0);
+                REQUIRE(matrix_iget(W, i, j) == matrix_iget(W0, i0, j0));
+            }
+        }
+        matrix_free(W);
+    }
+
+    matrix_free(W0);
+    bool_vector_free(ens_mask);
+    bool_vector_free(obs_mask);
+    ies_enkf_data_free(data);
+}
 
-    REQUIRE(matrix_iget(E_obs_deactivate, 0, 0) == 1.0);
-    REQUIRE(matrix_iget(E_obs_deactivate, 1, 0) == 3.0);
-    REQUIRE(matrix_iget(E_obs_deactivate, 2, 0) == 0.0);
-    REQUIRE(matrix_iget(E_obs_deactivate, 0, 1) == 0.0);
-    REQUIRE(matrix_iget(E_obs_deactivate, 1, 1) == 0.0);
-    REQUIRE(matrix_iget(E_obs_deactivate, 2, 1) == 0.0);
+TEST_CASE("ies_enkf_linalg_extract_active_A", "[analysis]") {
+    const int ens_size = 4;
+    const int obs_size = 10;
+    const int state_size = 10;
+    ies_enkf_data_type *data = (ies_enkf_data_type *)ies_enkf_data_alloc();
+    bool_vector_type *ens_mask = bool_vector_alloc(ens_size, true);
+    bool_vector_type *obs_mask = bool_vector_alloc(obs_size, true);
 
+    ies_enkf_config_set_ies_logfile(ies_enkf_data_get_config(data), "log");
+    ies_enkf_data_open_log(data);
+    ies_enkf_init_update(data, ens_mask, obs_mask, nullptr, nullptr, nullptr,
+                         nullptr, nullptr, nullptr);
+    ies_enkf_data_update_ens_mask(data, ens_mask);
+
+    matrix_type *A0 = matrix_alloc(state_size, ens_size);
+    for (int i = 0; i < state_size; i++) {
+        for (int j = 0; j < ens_size; j++)
+            matrix_iset(A0, i, j, i * ens_size + j);
+    }
+    ies_enkf_data_store_initialA(data, A0);
+
+    {
+        matrix_type *A = ies_enkf_alloc_activeA(data);
+        REQUIRE(matrix_equal(A, A0));
+        matrix_free(A);
+    }
+
+    // Deactivate one realization
+    bool_vector_iset(ens_mask, 1, false);
+    ies_enkf_data_update_ens_mask(data, ens_mask);
+    {
+        matrix_type *A = ies_enkf_alloc_activeA(data);
+        for (int i = 0; i < state_size; i++) {
```

Comment:
What this loop does is very complicated, is there any way of making it simpler? Personally would prefer:
```
for (int i = 0; i < state_size; i++) {
            for (int j = 0; j < ens_size - 1; j++) {
                int j0 = j;
                if(j > 0) {
                   j0++;
                }
                REQUIRE(matrix_iget(A, i, j) == matrix_iget(A0, i, j0));
            }
        }
```